users/yet-another-developer/process_records.c

```diff
@@ -0,0 +1,201 @@
+#include "yet-another-developer.h"
+
+uint16_t copy_paste_timer;
+
+__attribute__ ((weak))
+bool process_record_keymap(uint16_t keycode, keyrecord_t *record) {
+  return true;
+}
+
+__attribute__ ((weak))
+bool process_record_secrets(uint16_t keycode, keyrecord_t *record) {
+  return true;
+}
+
+// Defines actions for my global custom keycodes. Defined in yet-another-developer.h file
+// Then runs the _keymap's record handier if not processed here
+bool process_record_user(uint16_t keycode, keyrecord_t *record) {
+
+  // If console is enabled, it will print the matrix position and status of each key pressed
+#ifdef KEYLOGGER_ENABLE
+  #if defined(KEYBOARD_ergodox_ez) || defined(KEYBOARD_keebio_iris_rev2)
+    xprintf("KL: kc: %u, col: %u, row: %u, pressed: %u\n", keycode, record->event.key.row, record->event.key.col, record->event.pressed);
+  #else
+    xprintf("KL: kc: %u, col: %u, row: %u, pressed: %u\n", keycode, record->event.key.col, record->event.key.row, record->event.pressed);
+  #endif
+#endif //KEYLOGGER_ENABLE
+
+  switch (keycode) {
+  case KC_QWERTY ... KC_UNICODE:
+    if (record->event.pressed) {
+      set_single_persistent_default_layer(keycode - KC_QWERTY);
+    }
+    break;
+
+        case KC_MAKE:  // Compiles the firmware, and adds the flash command based on keyboard bootloader
+            if (!record->event.pressed) {
+                clear_mods();
+                clear_oneshot_mods();
+                send_string_with_delay_P(PSTR("make " QMK_KEYBOARD ":" QMK_KEYMAP), TAP_CODE_DELAY);
+                {
+                    #if defined(__arm__)
+                    send_string_with_delay_P(PSTR(":dfu-util"), TAP_CODE_DELAY);
+                    #elif defined(BOOTLOADER_DFU)
+                    send_string_with_delay_P(PSTR(":dfu"), TAP_CODE_DELAY);
+                    #elif defined(BOOTLOADER_HALFKAY)
+                    send_string_with_delay_P(PSTR(":teensy"), TAP_CODE_DELAY);
+                    #elif defined(BOOTLOADER_CATERINA)
+                    send_string_with_delay_P(PSTR(":avrdude"), TAP_CODE_DELAY);
+                    #endif // bootloader options
+                }
```

Comment:
This entire block can de replaced now.

All you need is:
```c
                    send_string_with_delay_P(PSTR(":flash"), TAP_CODE_DELAY);
```

This takes care of the detection.